components/x-privacy-manager/src/privacy-manager.jsx

```diff
@@ -0,0 +1,115 @@
+/**
+ * @typedef {{ ok: boolean, status?: number }} _Response
+ */
+
+import { h } from '@financial-times/x-engine';
+import { withActions } from '@financial-times/x-interaction';
+
+import s from './privacy-manager.scss';
+import { RadioBtn } from './radio-btn';
+import { LoadingMessage, ResponseMessage } from './messages';
+
+// TODO: Replace with the genuine endpoint
+export const CONSENT_API = 'https://consent.ft.com';
+
+export const withCustomActions = withActions(() => ({
+	onConsentChange() {
+		return ({ consent = true }) => ({ consent: !consent });
+	},
+
+	sendConsent() {
+		return async ({ consent }) => {
+			const body = JSON.stringify({
+				demographic: consent,
+				behavioural: consent,
+				programmatic: consent
+			});
+
+			try {
+				const res = await fetch(CONSENT_API, { method: 'PATCH', body });
+				return { _response: { ok: res.ok } };
+			} catch (err) {
+				return { _response: { ok: false } };
+			}
+		};
+	}
+}));
+
+/**
+ * @param {boolean} isLoading
+ * @param {_Response} response
+ * @param {string} referrer
+ */
+function renderMessage(isLoading, response, referrer) {
```

Comment:
Forgive me if I've not seen how this happens...

When a screen reader user submits this form, how do they know that a loading or response message has been shown?

If it's not already communicated, you can consider setting focus to the element or something like `aria-live`, but the best method will depend on more context than I have. Can catchup with you more if needed.

Reply:
That's an excellent point. I think `aria-live` is probably the right approach here.

I also think that a note on the Origami & X-Dash docs helping developers to be mindful of this scenario would be a fine PR for me to follow up with.

Reply:
I'd be interested in exploring this more with you. First question I'd ask is does this form need to be submitted via JavaScript or could there be a full page reload. Let me know.

Reply:
Unfortunately the App UI is stateful